### PR TITLE
Run downstream test with net-istio & mTLS

### PR DESCRIPTION
**Pull request by nak3**
This patch adds `test-e2e-with-mesh` for downstream test with net-istio&mTLS. 

servinge2e tests excludes the following tests for net-istio&mTLS:

- `servicemesh_test.go` ... test for kourier&istio
- `service_to_service_test.go` ... test for kourier&istio
- `verify_http_and_https_test.go` ... test for OpenShift Route's TLS. net-istio&mTLS does not use Route's TLS.
- `custom_route_test.go` ... test for OpenShift Route's escape hatch. Istio's Gateway is flexible and we don't need to support it for istio.
- `verify_route_conflict_test.go` ... needs to add extra namespace `conflict-test` and `test` to SMMR.

### CODE REVIEW

File: hack/lib/mesh_resources/smcp.yaml
Comment (skonto) [inline hack/lib/mesh_resources/smcp.yaml:14]: Can we add port 8444 here for metrics to work?

Comment (nak3) [reply hack/lib/mesh_resources/smcp.yaml:14]: Thank you @skonto I added `8444`.
By the way, do you have an idea how to stop alert error - [alerts.json](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-knative_serverless-operator/1085/pull-ci-openshift-knative-serverless-operator-main-4.7-upstream-e2e-aws-ocp-47/1412963876953133056/artifacts/upstream-e2e-aws-ocp-47/serverless-e2e/artifacts/build-Hddbxk3E/alerts.json).
We need to set  `metrics.backend-destination` to `"none"` like https://github.com/openshift-knative/serverless-operator/pull/1014 ?

**Comment by nak3** [comment]
/hold

@markusthoemmes @skonto @mgencur This is ready for review. Could you please take a look?

File: hack/lib/mesh_resources/smcp.yaml
Comment (skonto) [reply hack/lib/mesh_resources/smcp.yaml]: Nope we should have the target as is. The workaround that exposes the port should make things work.

**Comment by skonto** [comment]
As long as we dont lose coverage down some code path we are fine. I tested the setup locally.

/lgtm

File: hack/lib/mesh_resources/networkpolicy.yaml
Comment (markusthoemmes) [inline hack/lib/mesh_resources/networkpolicy.yaml:1]: Should we ship this together with Knative Serving maybe? Or are we going to document that users have to apply this manually?

File: hack/lib/serverless.bash
Comment (markusthoemmes) [inline hack/lib/serverless.bash:191]: YAY!

**Review by markusthoemmes — COMMENTED** [review]
LGTM on the surface. Have we verified that the "normal" test targets are not affected by the pkg move?

File: hack/lib/mesh_resources/networkpolicy.yaml
Comment (nak3) [reply hack/lib/mesh_resources/networkpolicy.yaml:1]: I think that it should be alright to ship it with Knative Serving (When istio ingress class is enabled, the networkpolicy is applied.).
@skonto are you alright? I think we need to update the doc https://docs.openshift.com/container-platform/4.6/serverless/serverless-release-notes.html#serverless-rn-1-16-0_serverless-release-notes as well.

File: openshift-knative-operator/cmd/operator/kodata/ingress/0.23/0-networkpolicy-mesh.yaml
Comment (markusthoemmes) [inline openshift-knative-operator/cmd/operator/kodata/ingress/0.23/0-networkpolicy-mesh.yaml:50]: This lacks the provider label, so it'll be applied even with kourier, which we don't want.

Comment (nak3) [reply openshift-knative-operator/cmd/operator/kodata/ingress/0.23/0-networkpolicy-mesh.yaml:50]: Thank you! Updated.

**Comment by nak3** [comment]
`Downstream test -> Upstream test` passes as [log](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-knative_serverless-operator/1085/pull-ci-openshift-knative-serverless-operator-main-4.7-upstream-e2e-aws-ocp-47/1420287465536622592)
`Upstream test -> Downstream test` fails due to subscription already exist [log](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-knative_serverless-operator/1085/pull-ci-openshift-knative-serverless-operator-main-4.7-operator-e2e-aws-ocp-47/1420287465465319424).

I will clean up the Makefile to run the mesh CI with `Downstream test -> Upstream test` order.

**Comment by skonto** [comment]
```
=== RUN   TestKafkaUserPermissions/edit_user
    user_permissions.go:51: Unexpected error creating bindings.knative.dev/v1beta1, Resource=kafkabindings, allowed = true, err = object is being deleted: kafkabindings.bindings.knative.dev "test-kafkabindings" already exists
    --- FAIL: TestKafkaUserPermissions/edit_user (0.10s)
```

**Comment by nak3** [comment]
/retest

Hmm... the error should not be related to this PR.
I opened https://github.com/openshift-knative/serverless-operator/pull/1123

**Comment by nak3** [comment]
/hold cancel
@markusthoemmes  I think this is ready to merge now.

**Review by markusthoemmes — APPROVED** [review]
/assign @maschmid 

LGTM! Want to give Marek the right of the last word.

**Pull request merged**
